### PR TITLE
[Chat][RTE] Styles update for send and message edit boxes

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
# What

- added styling for send box
- added edit box
- added a temporary edit box storybook page
- added styling for edit box
- moved common functionality
- fixed issue with always focused rich text editor

# Why

# How Tested

Storybook

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Comment by Leah-Xia-Microsoft** [comment]
Not sure whether this is within the scope of this PR but our placeholder text flicks when pressing the indent or bullet buttons while teams placeholder text will always displayed until there's some input.

https://github.com/Azure/communication-ui-library/assets/107075081/95b57e29-2396-4893-9ac0-18a95e8bff68

File: packages/react-components/src/components/ChatMessage/ChatMessageComponentAsRichTextEditBox.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+          ariaLabel={strings.editBoxCancelButton}
+          tooltipContent={strings.editBoxCancelButton}
+          onRenderIcon={onRenderThemedCancelIcon}
+          onClick={() => {
+            onCancel && onCancel(message.messageId);
+          }}
+          id={'dismissIconWrapper'}
+        />
+        <InputBoxButton
+          className={richTextActionButtonsStyle} //editingButtonStyle}
```
Comment (Leah-Xia-Microsoft) [inline packages/react-components/src/components/ChatMessage/ChatMessageComponentAsRichTextEditBox.tsx:118]: Should we remove the commented code?

File: packages/react-components/src/components/RichTextEditor/RichTextSendBox.tsx [hunk 2]
```diff
@@ -234,19 +265,16 @@ export const RichTextSendBox = (props: RichTextSendBoxProps): JSX.Element => {
...
-          className={sendButtonStyle}
-          ariaLabel={localeStrings.sendButtonAriaLabel}
-          tooltipContent={localeStrings.sendButtonAriaLabel}
-        />
-      </Stack.Item>
     </Stack>
   );
 };
+
+const sendBoxRichTextEditorStyle = (isExpanded: boolean): RichTextEditorStyleProps => {
```
Comment (Leah-Xia-Microsoft) [inline packages/react-components/src/components/RichTextEditor/RichTextSendBox.tsx:275]: Is it better to put this into the RichTextEditor.styles.ts?

**Review by Leah-Xia-Microsoft — APPROVED** [review]
Looks great!

**Comment by vhuseinova-msft** [comment]
> Not sure whether this is within the scope of this PR but our placeholder text flicks when pressing the indent or bullet buttons while teams placeholder text will always displayed until there's some input.
> 
>  Screen.Recording.2024-03-06.at.10.58.46.mov

It's RoosterJS issues. I've already posted them in
- https://github.com/microsoft/roosterjs/issues/2418
- https://github.com/microsoft/roosterjs/issues/2417

File: packages/react-components/src/components/ChatMessage/ChatMessageComponentAsRichTextEditBox.tsx (hunk 1 above)
Comment (vhuseinova-msft) [reply packages/react-components/src/components/ChatMessage/ChatMessageComponentAsRichTextEditBox.tsx:118]: Good catch, will remove!

File: packages/react-components/src/components/styles/SendBox.styles.ts [hunk 3]
```diff
@@ -96,14 +107,46 @@ export const borderAndBoxShadowStyle = (props: {
   hasErrorMessage: boolean;
   disabled: boolean;
 }): string => {
-  const { theme, hasErrorMessage, disabled } = props;
-  const borderColor = hasErrorMessage ? theme.semanticColors.errorText : theme.palette.neutralSecondary;
+  return mergeStyles(borderEditBoxStyle({ ...props, defaultBorderColor: props.theme.palette.neutralSecondary }));
+};
+
+/**
```
Comment (palatter) [inline packages/react-components/src/components/styles/SendBox.styles.ts:113]: Should we have a richTextSendBox Styles file?

File: packages/react-components/src/localization/locales/en-US/strings.json [hunk 4]
```diff
@@ -43,7 +43,8 @@
     "bulletListTooltip": "Bulleted list",
     "numberListTooltip": "Numbered list",
     "increaseIndentTooltip": "Increase indent",
-    "decreaseIndentTooltip": "Decrease indent"
+    "decreaseIndentTooltip": "Decrease indent",
+    "richTextFormatButtonTooltip": "Format"
```
Comment (palatter) [inline packages/react-components/src/localization/locales/en-US/strings.json:47]: Do we need richText prefix here?

Comment (vhuseinova-msft) [reply packages/react-components/src/localization/locales/en-US/strings.json:47]: I was thinking about `formatButtonTooltip` first but it wasn't very clear what is format here so I went with `richTextFormatButtonTooltip`. Any suggestions?

File: packages/react-components/src/components/styles/SendBox.styles.ts (hunk 3 above)
Comment (vhuseinova-msft) [reply packages/react-components/src/components/styles/SendBox.styles.ts:113]: It's not for send box per se but for the border component that's inside of input box component. Moved it to `RichTextEditor.styles` file

**Review by palatter — APPROVED** [review]

**Pull request merged**